src/wms/src/main/java/org/geoserver/wms/map/GetMapKvpRequestReader.java

```diff
@@ -764,6 +796,35 @@ private InputStream getHttpInputStream(URL styleUrl) throws IOException {
         }
     }
 
+    /** Executes the HTTP request with the max request time settings. */
+    private CloseableHttpResponse executeRequest(HttpCacheContext cacheContext, HttpGet httpget)
+            throws IOException, ClientProtocolException {
+        int hardTimeout = 60000;
+        String maxTimeStr =
+                GeoServerExtensions.getProperty(GETMAPKVPREQUESTREADER_MAXREQUEST_TIME_VARIABLE);
+        if (StringUtils.isNotEmpty(maxTimeStr)) {
```

Comment:
This code is almost a copy paste of the code above, but in one you check for a ``NULL`` value and here you checked for a ``NULL`` and emptiness. I would create an helper method and avoid the duplication.

Reply:
Makes sense, adding it.